### ISSUE TITLE
About Pytorchcv pre-trained mobilenetv2-w1 

### ISSUE THREAD

**Issue by feanor21115** [jakc4103/DFQ#15]
Hi, It is kind of you to share your work.
Very appreciate it.

I had used the model in the modeling folder. It works very well.
However, I've encountered some troubles when applying the method on Torchvision pre-trained MobilenetV2.

1. Both --absorption, and --correction,
    It seems that some layer in the graph don't have the attribute "bias".
    Actually, it got a "None".

2. I've skipped the problem above by if-else. 
    Result of the model in the modeling folder remains the same.  
    Back to Torchvision pre-trained MobilenetV2,
    it just like the closed issue #10 
    when --equalize, the accuracy drop tremendously to 0.1% 
    No matter with --absorption, --correction, --distill_range.

Do these outputs to be expected since specific methods in the implementation? 
If so, would you like to helping me finding out the reason?

Thank you.

**Comment by jakc4103** [comment]
what is the arguments you use when you got bias equals to None?
"python main.py --quantize --absorption --correction" ?

**Comment by feanor21115** [comment]
python main.py --quantize --relu --correction
python main.py --quantize --relu --equalize --absorption

when --absorption happens,
line 161 at dfq.py
graph[layer_second].bias.data.add_(wc)
and it comes up "nonetype does not have the attribute data"

so does --correction
line 287 at the same file
graph[idx_layer].bias.add_(-bias)
"nonetype does not have the attribute add_"

**Comment by jakc4103** [comment]
Worked on my PC.
Did you comment out merge_batchnorm in main_cls.py line 149 ?

**Comment by feanor21115** [comment]
No, I keep it there,
and merge_batchnorm generate fake_weight and fake_bias.

**Comment by jakc4103** [comment]
could you  check if the merge_batchnorm worked? (did it run into the if statement in utils/layer_transform.py line244)
if not, the model you use probably did not use torch.nn.Conv2d for convolution

**Comment by feanor21115** [comment]
OK, I will go check it.

**Comment by feanor21115** [comment]
Sorry, my bad.
Please forgive me for confusing torchvision and pytorchcv.

The model from 
import torchvision.models as models
model = models.mobilenet_v2(pretrained=True)
works very well.

However,
the model from
from pytorchcv.model_provider import get_model as ptcv_get_model
model = ptcv_get_model('mobilenetv2_w1', pretrained=True)
gets the problem 1.

and I have also check that both the model would entered the statement line244

**Comment by jakc4103** [comment]
Should be ok now with the latest commit.
The problem is that the final layer (classifier) did not use bias in pytorchcv model.

BTW, bias absorption is not working correctly in current implementation.

**Comment by feanor21115** [comment]
Thank you for helping me.
Appreciate it a lot!!!

**Comment by liming312** [comment]
Hi @jakc4103 , does bias absorption part work now?

**Comment by Gmxna** [comment]
@jakc4103 Hi, sorry to bring up this issue again, I also tried to replace the model with ptcv mobilenet model, and had the same problem of ACC around 0. The issue seems not to be solved, can you verify again? The change I made is replacing model with:
```
from pytorchcv.model_provider import get_model as ptcv_get_model
model = ptcv_get_model('mobilenetv2_w1', pretrained=True)
```
And run the script with `--equalize`
